source/extensions/filters/http/common/aws/utility.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "common/http/headers.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Aws {
+
+class Utility {
+public:
+  /**
+   * Creates a canonicalized header map according to
```

Comment:
Maybe explain why we are doing this; it's just for signing, right? We never put these on the wire..

Reply:
The whole canonicalization process is at the end of the day about consistently hashing HTTP requests. We do the exact same canonicalization upstream as part of our AuthN. This map is really just part of the input string to the hash we put in the Authorization header.

I'll explicitly say this is to be used as an input to the request canonicalization function. (Which I'll also move into this util)